driver/src/test/java/com/consol/citrus/db/driver/JdbcPreparedStatementTest.java

```diff
@@ -71,6 +76,23 @@ public void testSetParameterOverwritesValue() {
         Assert.assertEquals(jdbcPreparedStatement.getParameters().get("0"), 42);
     }
 
+    @Test
+    public void testAddParametersToBatch() throws SQLException {
+
+        //GIVEN
+        jdbcPreparedStatement.addBatch("SELECT id, name FROM cities WHERE abbreviation = ?");
```

Comment:
According to the spec, this should not be possible

```
 SQLException if a database access error occurs,
     * this method is called on a closed <code>Statement</code>, the
     * driver does not support batch updates, the method is called on a
     * <code>PreparedStatement</code> or <code>CallableStatement</code>
```

Reply:
According to the spec of JDBC 2.0:

```
public interface PreparedStatement extends Statement {

 ...

    //--------------------------JDBC 2.0-----------------------------

    /**
     * Adds a set of parameters to this <code>PreparedStatement</code>
     * object's batch of commands.
     *
     * @exception SQLException if a database access error occurs or
     * this method is called on a closed <code>PreparedStatement</code>
     * @see Statement#addBatch
     * @since 1.2
     */
    void addBatch() throws SQLException;

...
}
```

It states that one should add parameters to the objects batch commands...
But how would you add statements to the batch if `addBatch(String sql)` is not permitted? :thinking:

Reply:
:bulb: https://www.mkyong.com/jdbc/jdbc-preparedstatement-example-batch-update/